src/services/uniswapV3/UniswapV3Service.ts

```diff
@@ -0,0 +1,264 @@
+import {injectable} from 'inversify';
+import {Token, CurrencyAmount} from '@uniswap/sdk-core';
+import {FeeAmount, Pool, TickMath, tickToPrice} from '@uniswap/v3-sdk';
+import JSBI from 'jsbi';
+
+import {GraphTick, BarChartTick, TickProcessed} from './interfaces.js';
+import axios from 'axios';
+
+enum Direction {
+  ASC,
+  DESC,
+}
+
+const MAX_INT128 = JSBI.subtract(JSBI.exponentiate(JSBI.BigInt(2), JSBI.BigInt(128)), JSBI.BigInt(1));
+
+@injectable()
+export default class UniswapV3Service {
+  async getFullTickData(poolAddress: string): Promise<GraphTick[]> {
+    let allTicks: GraphTick[] = [];
+    let skip = 0;
+    let loadingTicks = true;
+
+    while (loadingTicks) {
+      const ticks = await this.getTickDataFromSubgraph(poolAddress, skip);
+      allTicks = allTicks.concat(ticks);
+      if (ticks.length < 1000) {
+        loadingTicks = false;
+      } else {
+        skip += 1000;
+      }
+    }
+
+    return allTicks;
+  }
+
+  async getTickDataFromSubgraph(poolAddress: string, skip: number): Promise<GraphTick[]> {
+    const ticksQuery = JSON.stringify({
+      query: `{ ticks(
+            where: {poolAddress: "${poolAddress.toLowerCase()}", liquidityNet_not: "0"}
+            first: 1000,
+            orderBy: tickIdx,
+            orderDirection: asc,
+            skip: ${skip}
+          ) {
+            tickIdx
+            liquidityGross
+            liquidityNet
+          }
+        }
+      `,
+    });
+
+    const response = await axios.post('https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v3', ticksQuery, {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+
+    return response.data.data.ticks;
+  }
+
+  async createBarChartTicks(
```

Comment:
this should be entry point for this service eg
```
apply(...) {
return this.createBarChartTicks(...)
}
```

we can keep original method name just for reference for original code

Reply:
does this method doing any external calls? or everything is calculated based on arguments?